design/kata-api-design.md

```diff
@@ -0,0 +1,143 @@
+# Kata API Design
+To fulfill the [kata design requirements](kata-design-requirements.md), and based on the disscusion on [Virtcontainers API extentions](https://docs.google.com/presentation/d/1dbGrD1h9cpuqAPooiEgtiwWDGCYhVPdatq7owsKHDEQ), the kata runtime library features following APIs:
+-  Sandbox based top API
+-  Storage and network hotplug API
+-  Plugin frameworks for external proprietary Kata runtime extensions
+-  Built-in shim and proxy types and capabilities
+
+## Sandbox Based API
```

Comment:
I don't disagree with this but I am just trying to go step by step here, and I don't think the naming should be a huge priority as it will need to be changed in a bunch of different places in the code. Changing the whole naming could be part of a second step, otherwise we're also going to have to change what is already there.

Reply:
I agree that the API set can be implemented in many-step PRs. But the implementation steps do not block the API set we want to define/provide.

And about the pod->sandbox renaming, there is a good reason to do it and we'd better do it sooner than later.

Reply:
Fair enough !

Reply:
I understand the reasoning but personally I really dislike the word "sandbox" as it's ambiguous:

- https://en.wikipedia.org/wiki/Sandbox_(computer_security)
- https://en.wikipedia.org/wiki/Sandbox_(software_development)

It might just be me, but when I think "sandbox" I invariably initially resolve that to mean "test area" before "security mechanism", which is wrong in this context of course.

Unless we want to go with a completely new noun like "vault", can I suggest that we add a brief definition of what we mean by "sandbox". This should mention that we're using the CRI terminology but also state that this is the equivalent of a k8s "pod" to avoid any confusion.